Packs/GigamonThreatINSIGHT/Integrations/GigamonThreatINSIGHT/GigamonThreatINSIGHT.py

```diff
@@ -0,0 +1,495 @@
+            if isinstance(r_json[data_type][item], dict):
+                r_json[data_type][item] = flattenDict(r_json[data_type][item])
+            if isinstance(r_json[data_type][item], list):
+                r_json[data_type][item] = flattenList(r_json[data_type][item])
+            new_pair = {item: str(r_json[data_type][item])}
+            new_item.update(new_pair)
+        data.append(new_item)
+        context[contextPath].append(createContext(new_item))
+    else:
+        for item in r_json[data_type]:
+            new_item = {}
+            for field in item:
+                # if there are any unicode characters, encode them to utf8
+                if isinstance(item[field], str):
+                    item[field] = str(item[field].encode("utf-8"))
+                new_pair = {field: str(item[field])}
+                new_item.update(new_pair)
+            data.append(new_item)
+            context[contextPath].append(createContext(new_item))
+    # submit results
+    demisto.results({'ContentsFormat': formats['table'], 'Type': entryTypes['note'], 'Contents': data, 'EntryContext': context})
+
+
+def main():
+    # get command and args
+    command = demisto.command()
+    args = getArgs()
+
+    # initialize common args
+    api_key = demisto.params().get('api_key')
+    account_uuid = demisto.params().get('account_uuid')
+    global HEADERS
+    HEADERS = {
+        'Authorization': 'IBToken ' + api_key,
+        'User-Agent': 'Cortex_Insight.v3',
+        'Content-Type': 'application/json',
+    }
+    # attempt command execution
+    try:
+        if command == 'test-module':
+            response = sendRequest('GET', 'Sensors', 'sensors')
+            demisto.results('ok')
+        if command == 'fetch-incidents':
+            # default first fetch to -7days
+            first_fetch_time = datetime.now() - timedelta(days=7)
+            max_results = arg_to_number(
+                arg=demisto.params().get('max_fetch'),
+                arg_name='max_fetch',
+                required=False
+            )
+
+            next_run, incidents = fetchIncidents(
+                account_uuid=account_uuid,
+                max_results=max_results,
+                last_run=demisto.getLastRun(),
+                first_fetch_time=first_fetch_time
+            )
+
+            demisto.setLastRun(next_run)
+            demisto.incidents(incidents)
+        elif command == 'insight-get-events':
+            if args['response_type'] == "metadata":
+                response_type = "metadata"
+            elif args['response_type'] == "aggregations":
+                pattern = r"^.*[Gg][Rr][Oo][Uu][Pp]\s+[Bb][Yy].*$"
+                if not re.search(pattern, args['query']):
+                    demisto.results("Error: No 'group by' statement in query. Aggregation requires a 'group by' statement.")
+                else:
+                    response_type = "aggregations"
+            else:
+                response_type = "events"
```

Comment:
1. Wrap every command with a function to make the code clear.
2. I didn't understand, if the pattern is not in the query argument do we want to raise an error?
3. Instead of using `responseToEntry` please use `return_results` following our convention. You can refer for more information [here](https://xsoar.pan.dev/docs/integrations/code-conventions#return_results:~:text=return%20results%2C%20here-,return_results,-%23).
```suggestion
        elif command == 'insight-get-events':
           if response_type := args.get('response_type'):
                if response_type == 'aggregations':
                    pattern = r"^.*[Gg][Rr][Oo][Uu][Pp]\s+[Bb][Yy].*$"
                    if not re.search(pattern, args.get('query', '')):
                        demisto.results("Error: No 'group by' statement in query. Aggregation requires a 'group by' statement.")
            else:
                response_type = "events"
```